cvc5_pythonic_api/cvc5_pythonic.py

```diff
@@ -1727,6 +1734,793 @@ def Or(*args):
     return _nary_kind_builder(Kind.OR, *args)
 
 
+#########################################
+#
+# Strings and Sequences
+#
+#########################################
+
+
+class SeqSortRef(SortRef):
+    def is_string(self):
+        """Determine if sort is a string
+
+        >>> s = StringSort()
+        >>> s.is_string()
+        True
+        >>> s = SeqSort(IntSort())
+        >>> s.is_string()
+        False
+        """
+        return isinstance(self, StringSortRef)
+
+
+class SeqRef(ExprRef):
+    """Sequence Expressions"""
+
+    def sort(self):
+        return _sort(self.ctx, self.ast)
+
+    def isSequence(self):
+        return True
+
+    def __add__(self, other):
+        """Concatenation of two sequences.
+
+        >>> s,t = Consts('s t',SeqSort(RealSort()))
+        >>> s+t
+        Concat(s, t)
+        >>> (s+t).sort()
+        (Seq Real)
+        """
+        return Concat(self, other)
+
+    def __radd__(self, other):
+        return Concat(other, self)
+
+    def __getitem__(self, i):
+        """Return the SMT expression `self[arg]`
+
+        >>> seq = Const('seq',SeqSort(IntSort()))
+        >>> seq[0]
+        seq [] 0
+        >>> seq[0].sort()
+        Int
+        """
+        if _is_int(i):
+            i = IntVal(i, self.ctx)
+        return _to_expr_ref(
+            self.ctx.solver.mkTerm(Kind.SEQ_NTH, self.ast, i.ast), self.ctx
+        )
+
+    def at(self, i):
+        """Return the sequence at index i
+
+        >>> seq = Const('seq',SeqSort(IntSort()))
+        >>> seq.at(0)
+        At(seq, 0)
+        >>> seq.at(0).sort()
+        (Seq Int)
+        """
+        if _is_int(i):
+            i = IntVal(i, self.ctx)
+        return SeqRef(self.ctx.solver.mkTerm(Kind.SEQ_AT, self.ast, i.ast), self.ctx)
+
+    def is_string(self):
+        return isinstance(self, StringRef)
+
+    def as_string(self):
+        """Return a string representation of sequence expression.
+
+        >>> s = Unit(IntVal(1)) + Unit(IntVal(2))
+        >>> s.as_string()
+        '(str.++ (seq.unit 1) (seq.unit 2))'
+        """
+        return str(self.ast)
```

Comment:
Could you give a doc-test where `self` is a value? For values, this function should return the python string which represents that value (rather than an SMT-LIB expression).